### PR TITLE
Update deps, add Scala 3 support, drop Scala 2.12

### PR DESCRIPTION
**Pull request by migesok**
Compared to conhub 1.2.1 the changes are source and binary compatible with these exceptions:
- potential source compat breakage: NamedDispatcher doesn't have an implicit argument in the constructor anymore but it appears to be unused even in the internal Evolution codebase

Changes highlights:
- updated plugins and dependencies
- added Scala 3.3.3 cross-compilation
- removed Scala 2.12 support
- introduced sbt-version-policy plugin and MiMa bincompat checks
- fixed compilation warnings, minor test refactoring
- replaced CurrentThreadExecutionContext usage with direct calls to ExecutionContext.parasitic since we do not support 2.12 anymore
- Scala 3 compiler complained about NamedDispatcher implicit arguments in the constructor. Since the class is unused even in the internal Evolution codebase, implicit modifier was removed and the type was marked as deprecated.

### CODE REVIEW

File: project/Dependencies.scala
Comment (edubrovski) [inline project/Dependencies.scala:14]: Not sure if it's a good idea to update scodec-bits from 1.1 to 1.2. It's not uncommon for Scala serialization libraries to change serialization format when they bump minor version.

**Review by edubrovski — CHANGES_REQUESTED** [review]

**Comment by edubrovski** [comment]
@migesok  Overall it looks like you're making the assumption that Scala libraries adhere to SEMVER and if major version hasn't changed then it's safe to bump. Unfortunately this is often not the case with Scala libraries. I propose to keep MRs small. If you need Scala 3 I think it's better to only update what's needed for Scala 3 in this MR/

File: project/Dependencies.scala
Comment (migesok) [reply project/Dependencies.scala:7]: 1.2.0 and 1.1.5 are source and binary compatible. Since MiMa was introduced only in 1.2.0, I'm not sure whether it extends to 1.1.4 but I believe it should be as the only change for 1.1.5 was Scala 3 introduction.

File: project/Dependencies.scala
Comment (migesok) [reply project/Dependencies.scala:19]: I believe everyone should be on 2.6.21 already, it doesn't seem to be a problem.

File: project/Dependencies.scala
Comment (migesok) [reply project/Dependencies.scala:14]: scodec-bits provides building block for making serializers, like "write me a byte" so it shouldn't break. But if you insist I can try to introduce binary format compatibility tests.

File: project/Dependencies.scala
Comment (migesok) [reply project/Dependencies.scala:8]: I added MiMa there:
https://github.com/evolution-gaming/akka-serialization/pull/146

1.1.0 is binary compatible with 1.0.5

File: src/main/scala/com/evolutiongaming/conhub/SequentialMapHelper.scala
Comment (edubrovski) [inline src/main/scala/com/evolutiongaming/conhub/SequentialMapHelper.scala:27]: Why do we need to change `ExecutionContext`? It this something we need for Scala 3?

**Comment by migesok** [comment]
@migesok Unfortunately it is not possible since dependencies also have to have Scala 3 support and many of them didn't.
They have to come in one set of changes. I can comment on individual dependency version changes if you wish.

File: src/main/scala/com/evolutiongaming/conhub/SequentialMapHelper.scala
Comment (edubrovski) [reply src/main/scala/com/evolutiongaming/conhub/SequentialMapHelper.scala:27]: Nvm, I checked the code in `execution-tools`, `CurrentThreadExecutionContext` simply delegates to `parasitic`, make sense

**Comment by edubrovski** [comment]
overall LGTM but I'd like to wait for one more person to approve before merging

File: project/Dependencies.scala
Comment (mr-git) [reply project/Dependencies.scala:19]: and most of our OSS base akka support libraries are on 2.6.21 for months already

Comment (vilunov) [reply project/Dependencies.scala:19]: No, we use .20 via overrides. The fact that our libraries are on .21 is a problem and not a reason for others to bump it too.

**Review by vilunov — CHANGES_REQUESTED** [review]

File: project/Dependencies.scala
Comment (mr-git) [reply project/Dependencies.scala:19]: I am afraid that it will sound arrogant, but prohibiting upgrades for all other projects because your one cannot or doesn't want to update is not a solution we can accept. 

You are free to put overrides in your project or to not update some library at your own risk.

Please review changes in akka (https://github.com/akka/akka/releases/tag/v2.6.21) - they are very small and, most probably, will not have negative affect on your project.

Comment (migesok) [reply project/Dependencies.scala:19]: I stand with @mr-git here - Akka 2.6.21 is more than 1 year old now. If there is an internal reason for us to use older versions internally, there is no reason to have an outdated version in the opensource. Updating Akka 19 and 20 to 21 is trivial and by just trying to hold it in transitive dependencies is a loosing game. There are already dependencies which will bring 2.6.21 transitively (like safe-akka). If you want safeguards, they should be on the library client side - dependency overrides and such.

**Review by edubrovski — APPROVED** [review]

**Review by edubrovski — CHANGES_REQUESTED** [review]
I'll approve after we resolve the discussion about Akka version

File: project/Dependencies.scala
Comment (dfakhritdinov) [reply project/Dependencies.scala:19]: @vilunov, am I right that we (including _you_) are using Akka `2.6.21` now? Shall we unblock the PR?

**Comment by migesok** [comment]
@vilunov confirmed that the issue has been resolved
I'm going ahead with the merge and release

**Pull request merged**
